libmatroska and libebml did not ship with PIC-enabled archives or shared
objects, and some distributions shipped them with lib(ebml|matroska)_pic name.
Portage versions are since a long time built in both .a and .so flavors, with
the same name. Now also upstream does this way so we're safe.

Index: vlc-0.8.4a/configure.ac
===================================================================
--- vlc-0.8.4a.orig/configure.ac
+++ vlc-0.8.4a/configure.ac
@@ -2026,16 +2026,9 @@ if test "${enable_mkv}" != "no" -a "${CX
               if test "${SYS}" = "darwin"; then
                 VLC_ADD_CXXFLAGS([mkv],[-O1])
               fi
-              AC_CHECK_LIB(ebml_pic, main, [
-                # We have ebml_pic, that's good, we can build an mkv.so plugin !
+              AC_CHECK_LIB(ebml, main, [
                 VLC_ADD_PLUGINS([mkv])
-                VLC_ADD_LDFLAGS([mkv],[-lmatroska_pic -lebml_pic])
-              ], [
-                AC_CHECK_LIB(ebml, main, [
-                  # We only have libebml, make mkv.a a builtin
-                  VLC_ADD_BUILTINS([mkv])
-                  VLC_ADD_LDFLAGS([mkv],[-lmatroska -lebml])
-                ])
+                VLC_ADD_LDFLAGS([mkv],[-lmatroska -lebml])
               ])
             ],
             [AC_MSG_RESULT([no])
